libp2p/dial.nim

```diff
@@ -69,5 +69,5 @@ method addTransport*(
 method tryDial*(
   self: Dial,
   peerId: PeerId,
-  addrs: seq[MultiAddress]): Future[MultiAddress] {.async, base.} =
+  addrs: seq[MultiAddress]): Future[Opt[MultiAddress]] {.async, base.} =
```

Comment:
when using opt, don't forget to import/export `results` - it might work now, but if a module you depend on stops exporting results, this is break - always import what is used directly in the module and export that which appears in public interfaces

Reply:
Thanks for letting me know. I've pushed a new commit.